js/katex-a11y.js

```diff
@@ -0,0 +1,464 @@
+                buildA11yStrings(tree.value.index, a11yStrings);
+                a11yStrings.push("end index");
+            }
+
+            a11yStrings.push("square root of");
+            buildA11yStrings(tree.value.body, a11yStrings);
+            a11yStrings.push("end square root");
+        });
+    },
+
+    supsub: function(tree, a11yStrings) {
+        if (tree.value.base) {
+            buildA11yStrings(tree.value.base, a11yStrings);
+        }
+
+        if (tree.value.sub) {
+            buildRegion(a11yStrings, function(a11yStrings) {
+                a11yStrings.push("start subscript");
+                buildA11yStrings(tree.value.sub, a11yStrings);
+                a11yStrings.push("end subscript");
+            });
+        }
+
+        if (tree.value.sup) {
+            // There are some cases that just read better if we don't have
+            // the extra start/end baggage, so we skip the extra text
+            var newPower = (powerMap[tree.value.sup] ||
+                tree.value.sup.value && tree.value.sup.value[0] &&
+                powerMap[tree.value.sup.value[0].value] ||
+                tree.value.sup.value &&
+                powerMap[tree.value.sup.value]);
+
+            buildRegion(a11yStrings, function(a11yStrings) {
+                if (newPower) {
+                    a11yStrings.push(newPower);
+                    return;
+                }
+
+                a11yStrings.push("start superscript");
+                buildA11yStrings(tree.value.sup, a11yStrings);
+                a11yStrings.push("end superscript");
+            });
+        }
+    },
+
+    text: function(tree, a11yStrings) {
+        if (typeof tree.value !== "string") {
+            buildA11yStrings(tree.value.body, a11yStrings);
+        } else {
+            buildString(tree, "normal", a11yStrings);
+        }
+    },
+
+    textord: function(tree, a11yStrings) {
+        buildA11yStrings(tree.value, a11yStrings);
+    }
+};
+
+var buildA11yStrings = function(tree, a11yStrings) {
+    a11yStrings = a11yStrings || [];
+
+    // Handle strings
+    if (typeof tree === "string") {
+        buildString(tree, "normal", a11yStrings);
+
+    // Handle arrays
+    } else if (tree.constructor === Array) {
+        for (var i = 0; i < tree.length; i++) {
+            buildA11yStrings(tree[i], a11yStrings);
+        }
+
+    // Everything else is assumed to be an object...
+    } else {
+        if (!tree.type || !(tree.type in typeHandlers)) {
+            throw "KaTeX a11y un-recognized type: " + tree.type;
+        } else {
+            typeHandlers[tree.type](tree, a11yStrings);
+        }
+    }
+
+    return a11yStrings;
+};
+
+var renderStrings = function(a11yStrings, a11yNode) {
+    var doc = a11yNode.ownerDocument;
+
+    for (var i = 0; i < a11yStrings.length; i++) {
+        var a11yString = a11yStrings[i];
+
+        if (i > 0) {
+            a11yNode.appendChild(doc.createTextNode(", "));
```

Comment:
Do we add commas here to make the screenreader put space in between the words or something? Might want to mention why in a comment.